src/util/createServerRootMixin.js

```diff
@@ -88,36 +134,35 @@ function augmentInstantSearch(
    * @returns {Promise} result of the search, to save for .hydrate
    */
   search.findResultsState = function(componentInstance) {
-    let _renderToString;
-    try {
-      _renderToString = require('vue-server-renderer/basic');
-    } catch (e) {
-      // error is handled by regular if, in case it's `undefined`
-    }
-    if (!_renderToString) {
-      throw new Error('you need to install vue-server-renderer');
-    }
-
     let app;
+    let renderedApp;
 
     return Promise.resolve()
       .then(() => {
-        app = cloneComponent(componentInstance);
-
-        app.instantsearch.helper = helper;
-        app.instantsearch.mainHelper = helper;
-
-        app.instantsearch.mainIndex.init({
-          instantSearchInstance: app.instantsearch,
-          parent: null,
-          uiState: app.instantsearch._initialUiState,
+        app = cloneComponent(componentInstance, {
```

Comment:
Unlike vue 2, in vue 3, we cannot access its instance variables like this.
So in order to manipulate its data, I'm passing a mixin to `cloneComponent`, so that the manipulation can happen in `created` lifecycle.